### PR TITLE
Multi layer: diagnostics, documentations, and testing.

### PR DESCRIPTION
**Pull request by crocha700**
This also adds documentation regarding the multi-layer model and its diagnostics, adds a test using the multi-layer model and the same parameters as the two-layer model test, and   **fixes the bug** described in #86

### CODE REVIEW

**Comment by rabernat** [comment]
@crocha700  - Can you make the output for the logging less verbose so that the test output is easier to read?

File: pyqg/test_bt.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import datetime as dt
+import numpy as np
+import matplotlib.pyplot as plt
```
Comment (rabernat) [inline pyqg/test_bt.py]: We should not import from matplotlib in any of the tests.

**Comment by rabernat** [comment]
@crocha700 - it's great you are putting so much work into this! I have to admit that these huge commits are very hard to review. There are at least 5 completely separate features added here, and your last one was the same. Maybe for the future, consider splitting things up into multiple smaller PRs. (I recognize that this might be difficult when all the features are interdependent.)

File: pyqg/kernel.pyx [hunk 2]
```diff
@@ -295,7 +295,7 @@ cdef class PseudoSpectralKernel:
         self.ifft_qh_to_q()
         # input might have been destroyed, have to re-assign
         self.qh[:] = b_view
-        
+
```
Comment (rabernat) [inline pyqg/kernel.pyx]: The fact that you are mixing whitespace cleanup with major changes to the core computational routines makes the code quite hard to review.

Comment (crocha700) [inline pyqg/kernel.pyx]: Yes, this is really annoying. Sorry. Will do a much bette job next time.

**Comment by crocha700** [comment]
Ok, that test_bt was an old thing that I added accidentally in this commit. Should pass everything now. Also logger looks better now.

**Comment by crocha700** [comment]
OK, I'm merging this into develop. This was my last changes in the layered-model, which should be working OK. Please test it out if you have any experiments in mind.

**Comment by rabernat** [comment]
No! please don't merge without updating whats-new.rst in the docs.

**Comment by crocha700** [comment]
@rabernat does the documentation builds automatically in readthedocs?

**Comment by rabernat** [comment]
Too late again...

**Comment by crocha700** [comment]
Sorry. OK, next time I'll assign somebody to review/merge to avoid this confusion again...

**Comment by rabernat** [comment]
The documentation is here:
http://pyqg.readthedocs.org/en/develop/examples.html

You can check the build status here:
https://readthedocs.org/projects/pyqg/builds/

**Comment by rabernat** [comment]
@crocha700, I'm very happy you are working so hard on this. The last thing I want to do is slow you down or discourage you.

My main request is that you keep your PRs more "atomic", i.e. one feature change per PR, in the future. This will make them easier to review. You might also want to consider [rebasing your PRs](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request) when they contain tons of individual commits.

**Comment by crocha700** [comment]
Fair enough. I'm actually rebasing my other branch "Eady model", which implements the stacked Eady model with the classic Eady model as a special case. I promise that that PR will be smoother.

**Comment by crocha700** [comment]
@rabernat I just learned that we can easily revert merged PRs (there's a button called revert just next to the merged message). Would it be useful to revert this?

**Comment by rabernat** [comment]
Nope, just create a new PR with the whats-new updates. I think it is important to keep track of the changes we are making in the docs.

-R

**Pull request merged**